crates/adapters/src/format/avro/deserializer.rs

```diff
@@ -459,9 +383,13 @@ impl<'de> de::Deserializer<'de> for &'_ Deserializer<'de> {
     where
         V: Visitor<'de>,
     {
-        match unwrap_union(self.input) {
-            Value::Record(ref fields) => visitor.visit_map(RecordDeserializer::new(fields)),
-            Value::Null => visitor.visit_map(RecordDeserializer::new(&[])),
+        let (value, schema) = unwrap_union(self.input, self.schema);
+
+        match value {
+            Value::Record(ref fields) => {
+                visitor.visit_map(RecordDeserializer::new(fields, schema)?)
+            }
+            Value::Null => visitor.visit_map(RecordDeserializer::new(&[], schema)?),
```

Comment:
map? in other places it's null.

Reply:
Stole this from serde_avro. I think the idea is that Null should behave like an empty map in this case.

Reply:
a null map is not an empty map.
this looks wrong

Reply:
map values are nullable in our implementation

Reply:
I removed this clause. We deal with nullable maps separately, so this may not be needed.